### DISCUSSION TITLE
Had hard times with `firewalld` and Docker

### DISCUSSION THREAD

**Discussion by sneko** [coollabsio/coolify#3433] [Improvement Requests]
Hi,

I wanted to rely on `firewalld` instead of directly with `iptables` (so I disabled it from Docker). No egress traffic was going out of Docker for whatever reason.

On my VPS I did all my tests with: `docker run --rm curlimages/curl https://curl.se/`

The only thing to make it working was:
```shell
sudo firewall-cmd --permanent --zone=docker --add-interface=docker0
firewall-cmd --permanent --zone=public --add-rich-rule='rule family="ipv4" source address=172.17.0.0/16 masquerade'
firewall-cmd --reload
```

But I struggled understanding why all my containers managed by Coolify didn't benefit from this fix. In fact all Coolify containers are in a dedicated network (that is not the `bridge` network, the default from Docker). And it uses a random interface `br-xxxx` that you can find with `ip link show`. It has some differences over the `docker0` interface used by `bridge` (information like IP range can be found by using `docker network inspect bridge` and `docker network inspect coolify`).

So the solution was to do the same for the other interface:
```shell
sudo firewall-cmd --permanent --zone=docker --add-interface=br-xxxxxx
firewall-cmd --permanent --zone=public --add-rich-rule='rule family="ipv4" source address=172.18.0.0/16 masquerade'
firewall-cmd --reload
```

And it's working! I could verified it with `docker run --rm --network coolify curlimages/curl https://curl.se/`.

I post this for record because I lost a few hours struggling with this. I don't know if @andrasbacsai you are more expert with `firewalld`, but maybe having your thoughts, or an example how to do it better would help? Maybe Coolify could have a fixed interface name so it's easy by default to preconfigure the server?

_PS: I posted on https://forums.docker.com/t/docker-and-firewalld-could-not-resolve-host-from-my-containers/143762 but the bot hid my reply with context and some step-by-step solution._

**Comment by sneko** [comment]
Just to add that at the end I stopped using `firewalld` due to race condition of services when having `cloud-init` on the server, there is no official proposal so I felt uncomfortable using it: https://github.com/firewalld/firewalld/issues/414

**Comment by Talkless** [comment]
@sneko I see you used `--add-interface=docker0`, but Coolify seems to create some random `br-7da043d4db4d` bridge, while `docker0` is `DOWN` in my case.

**Reply by sneko** [reply]
Hi @Talkless , I don't remember about the whole thing, but in my post I did mention the usage of the random interface `br-xxxx` as part of the solution for Coolify.

I don't know since then if there is a plan to make it "constant" so it's easy to script 🤷

**Reply by Talkless** [reply]
>  but in my post I did mention the usage of the random interface

Missed  that, sorry.

> I don't know since then if there is a plan to make it "constant" so it's easy to script 🤷

Yeah I've made feature request: https://github.com/coollabsio/coolify/discussions/5893

Also, ip addresses of containers are random too, so managing dna'ting manually is kinda "imposibru"?

**Reply by Talkless** [reply]
Oh there's `docker-proxy` that listens on host itself, "forwarding" traffic into containers, so there's no really need to do dnat'ing, unless you have issues with `docker-proxy` (and I see people complaining here and there)...
